modules/core/src/convert.dispatch.cpp

```diff
@@ -277,9 +277,19 @@ void Mat::convertTo(OutputArray dst, int type_, double alpha, double beta) const
     int cn = channels();
     int dtype = CV_MAKETYPE(ddepth, cn);
 
+    // refcount add 1
     Mat src = *this;
-    dst.create(dims, size, dtype);
-    Mat dstMat = dst.getMat();
+    Mat dstMat;
+    int refcount = this->u->refcount;
+    if (this->data == dst.getMat().data && dst.getMat().u != NULL && this->elemSize() == CV_ELEM_SIZE(dtype) && refcount == 2) {
```

Comment:
it's quite bad idea to do it at such a high-level with such unsafe low-level details: 1) refcount==2, 2) 3 calls to getMat(), which can be rather expensive. 3) dst.assign() call, which is not that safe.

Reply:
@vpisarev  I've added a function for type conversion, but it seems getMat needs to be called twice.

Reply:
@vpisarev 
I have added a reinterpretType API to reinterpret the data type of Mat. However, it seems that these checks are unavoidable. Instead, I have changed it to refcount == 1 and reduced the number of dst.getMat calls.